modules/core/src/main/java/org/apache/ignite/spi/systemview/view/ComputeJobView.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.spi.systemview.view;
+
+import java.util.StringJoiner;
+import java.util.UUID;
+import org.apache.ignite.internal.managers.systemview.walker.Order;
+import org.apache.ignite.internal.processors.cache.distributed.dht.GridReservable;
+import org.apache.ignite.internal.processors.job.GridJobProcessor;
+import org.apache.ignite.internal.processors.job.GridJobWorker;
+import org.apache.ignite.lang.IgniteUuid;
+
+/**
+ * Compute job representation for a {@link SystemView}.
+ */
+public class ComputeJobView {
+    /** Job. */
+    public final GridJobWorker job;
+
+    /** Job id. */
+    public final IgniteUuid id;
+
+    /**
+     * @param id Job id.
+     * @param job Job.
+     */
+    public ComputeJobView(IgniteUuid id, GridJobWorker job) {
+        this.id = id;
+        this.job = job;
+    }
+
+    /** @return Job id. */
+    @Order
+    public IgniteUuid id() {
+        return id;
+    }
+
+    /**
+     * {@link ComputeJobView#sessionId()} value equal to the value of {@link ComputeTaskView#sessionId()}
+     * if both records represents parts of the same computation.
+     *
+     * @see ComputeTaskView#sessionId()
+     * @return Session id.
+     */
+    @Order(1)
+    public IgniteUuid sessionId() {
+        return job.getSession().getId();
+    }
+
+    /** @return Origin node id. */
+    @Order(2)
+    public UUID originNodeId() {
+        return job.getTaskNode().id();
+    }
+
+    /** @return Task name. */
+    @Order(3)
+    public String taskName() {
+        return job.getSession().getTaskName();
+    }
+
+    /** @return Task class name. */
+    @Order(4)
+    public String taskClassName() {
+        return job.getSession().getTaskClassName();
+    }
+
+    /** @return Affinity cache name or {@code null} for non affinity call. */
```

Comment:
Comma separated list of affinity cache identifiers?

Reply:
Yes. Thank you. Fixed.